core/src/main/java/org/infinispan/interceptors/locking/ClusteringDependentLogic.java

```diff
@@ -128,15 +129,19 @@ protected abstract void commitSingleEntry(CacheEntry entry, Metadata metadata, F
 
       protected abstract WriteSkewHelper.KeySpecificLogic initKeySpecificLogic(boolean totalOrder);
 
-      protected void notifyCommitEntry(boolean created, boolean removed, CacheEntry entry, InvocationContext ctx,
-                                       FlagAffectedCommand command, Object previousValue, Metadata previousMetadata) {
+      protected void notifyCommitEntry(boolean created, boolean removed, boolean expired, CacheEntry entry,
+              InvocationContext ctx, FlagAffectedCommand command, Object previousValue, Metadata previousMetadata) {
          boolean isWriteOnly = (command instanceof WriteCommand) && ((WriteCommand) command).isWriteOnly();
          if (removed) {
             if (command instanceof RemoveCommand) {
                ((RemoveCommand)command).notify(ctx, previousValue, previousMetadata, false);
             } else {
-               notifier.notifyCacheEntryRemoved(
-                  entry.getKey(), previousValue, previousMetadata, false, ctx, command);
+               if (expired) {
+                  notifier.notifyCacheEntryExpired(entry.getKey(), previousValue, previousMetadata, ctx);
```

Comment:
Yeah this was something I found out is that eviction & invalidation notifications don't work properly on a tx cache, heh.